localstack/services/dynamodb/provider.py

```diff
@@ -498,37 +501,36 @@ def delete_table(self, context: RequestContext, table_name: TableName) -> Delete
 
         return result
 
-    def _forward_request(self, context: RequestContext, region: str | None) -> ServiceResponse:
-        """This helper is used to modify the request context for operations that make use of a global table region."""
-        if region:
-            with modify_context_region(context, region):
-                return self.forward_request(context)
-        return self.forward_request(context)
-
     def describe_table(self, context: RequestContext, table_name: TableName) -> DescribeTableOutput:
-        global_table_region = find_global_table_region(context, table_name)
-        # Check if table exists, to avoid error log output from DynamoDBLocal
-        if (
-            global_table_region
-            and not self.table_exists(context.account_id, global_table_region, table_name)
-        ) or not self.table_exists(context.account_id, context.region, table_name):
-            raise ResourceNotFoundException("Cannot do operations on a non-existent table")
```

Comment:
This logic is moved to refactored `get_global_table_region` to improve readability. See below.